packages/rich-text/src/plugins/Quote/index.tsx

```diff
@@ -48,60 +54,65 @@ const createBlockQuote = (editor: SPEditor) => {
   }
 };
 
-export function withQuoteEvents(editor: SPEditor, event: KeyboardEvent) {
-  if (!editor.selection) return;
-
-  const [currentFragment] = Editor.fragment(editor, editor.selection.focus.path) as CustomElement[];
-  const isEnter = event.keyCode === 13;
-
-  if (isEnter && currentFragment?.type === BLOCKS.QUOTE) {
-    event.preventDefault();
-
-    const text = { text: '' };
-    const paragraph = { type: BLOCKS.PARAGRAPH, children: [text] };
-
-    if (hasSelectionText(editor)) {
-      const currentOffset = editor.selection.focus.offset;
-      const currentTextLength = Node.string(currentFragment).length;
-      const cursorIsAtTheBeginning = currentOffset === 0;
-      const cursorIsAtTheEnd = currentTextLength === currentOffset;
-
-      if (cursorIsAtTheBeginning) {
-        Transforms.insertNodes(editor, paragraph, { at: editor.selection });
-      } else if (cursorIsAtTheEnd) {
-        Transforms.insertNodes(editor, paragraph);
+export function withQuoteEvents(editor: SPEditor) {
+  return (event: KeyboardEvent) => {
+    if (!editor.selection) return;
+
+    const [currentFragment] = Editor.fragment(
+      editor,
+      editor.selection.focus.path
+    ) as CustomElement[];
+    const isEnter = event.keyCode === 13;
+
+    if (isEnter && currentFragment?.type === BLOCKS.QUOTE) {
+      event.preventDefault();
+
+      const text = { text: '' };
+      const paragraph = { type: BLOCKS.PARAGRAPH, children: [text] };
+
+      if (hasSelectionText(editor)) {
+        const currentOffset = editor.selection.focus.offset;
+        const currentTextLength = Node.string(currentFragment).length;
+        const cursorIsAtTheBeginning = currentOffset === 0;
+        const cursorIsAtTheEnd = currentTextLength === currentOffset;
+
+        if (cursorIsAtTheBeginning) {
+          Transforms.insertNodes(editor, paragraph, { at: editor.selection });
+        } else if (cursorIsAtTheEnd) {
+          Transforms.insertNodes(editor, paragraph);
+        } else {
+          // Otherwise the cursor is in the middle
+          Transforms.splitNodes(editor);
+          Transforms.setNodes(editor, paragraph);
+        }
       } else {
-        // Otherwise the cursor is in the middle
-        Transforms.splitNodes(editor);
         Transforms.setNodes(editor, paragraph);
       }
-    } else {
-      Transforms.setNodes(editor, paragraph);
     }
-  }
 
-  const isBackspace = event.keyCode === 8;
-  const isMod = event.ctrlKey || event.metaKey;
-  const isShift = event.shiftKey;
-  const isOneKey = event.keyCode === 49;
+    const isBackspace = event.keyCode === 8;
```

Comment:
I'm wondering if we can override some SlateJS methods such as `deleteBackward`, `deleteForward` or `deleteFragment` instead of checking the backspace key here.